File: australia/australia/report/gst_audit_report/gst_audit_report.js
// Copyright (c) 2016, DAS and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["GST Audit Report"] = {
	"onload":function(){
		frappe.query_report.filters[2].$wrapper.hide()
		frappe.query_report.filters[1].$wrapper.hide()

		frappe.query_report.filters[0].$input.on("click",function(){
			if (frappe.query_report.get_filter_value('transaction_type') == "Sales"){
				frappe.query_report.filters[1].$wrapper.show()
				frappe.query_report.filters[2].$wrapper.hide()
			} else if (frappe.query_report.get_filter_value('transaction_type') == "Purchase"){
				frappe.query_report.filters[2].$wrapper.show()
				frappe.query_report.filters[1].$wrapper.hide()
			}
		})
	},
	"filters": [
		{
			fieldname:"transaction_type",
			label: __("Transaction Type"),
			fieldtype: "Select",
			options: ["Sales", "Purchase"],
			reqd:1
		},
		{
			fieldname:"sales_gst_type",
			label: __("Sales GST Type"),
			fieldtype: "Select",
			options: ["GST on Income - Non Export", "GST on Income - Export", "GST Free Income", "GST Free Exports", "Input Taxed", "BAS Excluded"],	
			default: "GST on Income - Non Export"
		},
		{
			fieldname:"purchase_gst_type",
			label: __("Purchase GST Type"),
			fieldtype: "Select",
			options: ["GST on Capital", "GST on Expenses", "GST Free Capital", "Input Taxed", "GST Free Expenses", "BAS Excluded"],
			default: "GST on Capital"
		},
	]
};

File: australia/australia/report/gst_audit_report/gst_audit_report.py
# Copyright (c) 2013, DAS and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
	columns, data = [], []
	doctype = ""
	where = ""
	party = ""
	
	if filters.get("transaction_type") == "Sales":
		doctype = "Sales Invoice"
		party = "customer"
	elif filters.get("transaction_type") == "Purchase":
		doctype = "Purchase Invoice"
		party = "supplier"

	if filters.get("transaction_type") == "Sales" and filters.get("sales_gst_type"):
		where = """ AND sales_gst_type = "{0}" """.format(filters.get("sales_gst_type"))
	elif filters.get("transaction_type") == "Purchase" and filters.get("purchase_gst_type"):
		where = """ AND purchase_gst_type = "{0}" """.format(filters.get("purchase_gst_type"))
	
	columns = [
			_("Date")+":Date:120",
			_("Document No")+":Data:200",
			_("Party")+":Data:180",
			_("Total")+":Currency:150",
			_("Taxes and Charges")+":Currency:150",
			_("Grand Total")+":Currency:150"	
		]
	if doctype:
		data = frappe.db.sql("""
			SELECT 
			posting_date, name, {0}, total, total_taxes_and_charges, grand_total
			FROM `tab{1}` 
			WHERE docstatus = 1 {2}
			""".format(party,doctype,where))

	return columns, data
